packages/core/lib/repl.js

```diff
@@ -49,16 +49,11 @@ ReplManager.prototype.start = function(options) {
       // If we exit for some reason, call done functions for good measure
       // then ensure the process is completely killed. Once the repl exits,
       // the process is in a bad state and can't be recovered (e.g., stdin is closed).
-      var doneFunctions = self.contexts.map(function(context) {
-        return context.done
-          ? function() {
-              context.done();
-            }
-          : function() {};
-      });
-      async.series(doneFunctions, function() {
-        process.exit();
-      });
+      const doneFunctions = self.contexts.reduce((fns, ctx) => {
```

Comment:
Since part of this work is improving readability,
I'd generally prefer callback arguments to be spelled out.

```suggestion
      const doneFunctions = self.contexts.reduce((functions, context) => {
```

Reply:
I also dig the suggestion [here] (https://github.com/trufflesuite/truffle/pull/2972#issuecomment-614998441,
although the `reduce` is pretty readable as well (and an improvement from the existing code).